app/routes.py

```diff
@@ -1 +1,196 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from datetime import datetime
+import os, requests
+
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+def validate_item_by_id(model, id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message":f"{model.__name__} {id} invalid"}, 400))
+
+    item = model.query.get(id)
+
+    return item if item else abort(make_response({"message":f"{model.__name__} {id} not found"}, 404))
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body:
+        abort(make_response({"details":"Invalid data"}, 400))
+
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"])
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    query = request.args.get("sort")
+
+    if query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+    return jsonify(tasks_response)
+
+
+@tasks_bp.route("/<id>", methods=["GET"])
+def get_one_task(id):
+    task = validate_item_by_id(Task, id)
+    return {"task": task.to_dict()}, 200
+
+
+@tasks_bp.route("/<id>", methods=["PUT"])
+def update_task(id):
+    task = validate_item_by_id(Task, id)
+
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    
+    db.session.commit()
+    return {"task": task.to_dict()}, 200
+
+
+@tasks_bp.route("/<id>", methods=["DELETE"])
+def delete_task(id):
+    task = validate_item_by_id(Task, id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return {
+        "details": f"Task {id} \"{task.title}\" successfully deleted"
+    }, 200
+
+
+@tasks_bp.route("/<id>/mark_complete", methods=["PATCH"])
+def mark_complete(id):
+    task = validate_item_by_id(Task, id)
+
+    task.completed_at = datetime.utcnow()
+
+    url = "https://slack.com/api/chat.postMessage"
```

Comment:
Consider making the `URL` into a constant since this endpoint probably won't change inside this function!

Soon, you'll work on projects with multiple API calls that need to happen, and you'll be working with multiple keys. Consider renaming these variable to be more specific like `SLACK_URL` and `slack_headers`.